// @ng-dynamic-forms/ui-material/esm2015/src/form-group/dynamic-material-form-group.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, EventEmitter, Input, Output } from "@angular/core";
import { FormGroup } from "@angular/forms";
import { DynamicFormControlComponent, DynamicFormGroupModel, DynamicFormLayoutService, DynamicFormValidationService } from "@ng-dynamic-forms/core";
export class DynamicMaterialFormGroupComponent extends DynamicFormControlComponent {
    /**
     * @param {?} layoutService
     * @param {?} validationService
     */
    constructor(layoutService, validationService) {
        super(layoutService, validationService);
        this.layoutService = layoutService;
        this.validationService = validationService;
        this.bindId = true;
        this.blur = new EventEmitter();
        this.change = new EventEmitter();
        this.customEvent = new EventEmitter();
        this.focus = new EventEmitter();
    }
}
DynamicMaterialFormGroupComponent.decorators = [
    { type: Component, args: [{
                selector: "dynamic-material-form-group",
                template: "<ng-container [formGroup]=\"group\">\n\n    <div role=\"group\"\n         [dynamicId]=\"bindId && model.id\"\n         [formGroupName]=\"model.id\"\n         [ngClass]=\"[getClass('element', 'control'), getClass('grid', 'control')]\">\n\n        <dynamic-material-form-control *ngFor=\"let _model of model.group\"\n                                       [group]=\"control\"\n                                       [hidden]=\"_model.hidden\"\n                                       [layout]=\"layout\"\n                                       [model]=\"_model\"\n                                       [templates]=\"templates\"\n                                       [ngClass]=\"[getClass('element', 'host', _model), getClass('grid', 'host', _model)]\"\n                                       (blur)=\"onBlur($event)\"\n                                       (change)=\"onChange($event)\"\n                                       (focus)=\"onFocus($event)\"\n                                       (matEvent)=\"onCustomEvent($event, null, true)\"></dynamic-material-form-control>\n    </div>\n\n</ng-container>"
            }] }
];
/** @nocollapse */
DynamicMaterialFormGroupComponent.ctorParameters = () => [
    { type: DynamicFormLayoutService, },
    { type: DynamicFormValidationService, },
];
DynamicMaterialFormGroupComponent.propDecorators = {
    "bindId": [{ type: Input },],
    "group": [{ type: Input },],
    "layout": [{ type: Input },],
    "model": [{ type: Input },],
    "templates": [{ type: Input },],
    "blur": [{ type: Output },],
    "change": [{ type: Output },],
    "customEvent": [{ type: Output },],
    "focus": [{ type: Output },],
};
function DynamicMaterialFormGroupComponent_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    DynamicMaterialFormGroupComponent.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    DynamicMaterialFormGroupComponent.ctorParameters;
    /** @type {!Object<string,!Array<{type: !Function, args: (undefined|!Array<?>)}>>} */
    DynamicMaterialFormGroupComponent.propDecorators;
    /** @type {?} */
    DynamicMaterialFormGroupComponent.prototype.bindId;
    /** @type {?} */
    DynamicMaterialFormGroupComponent.prototype.group;
    /** @type {?} */
    DynamicMaterialFormGroupComponent.prototype.layout;
    /** @type {?} */
    DynamicMaterialFormGroupComponent.prototype.model;
    /** @type {?} */
    DynamicMaterialFormGroupComponent.prototype.templates;
    /** @type {?} */
    DynamicMaterialFormGroupComponent.prototype.blur;
    /** @type {?} */
    DynamicMaterialFormGroupComponent.prototype.change;
    /** @type {?} */
    DynamicMaterialFormGroupComponent.prototype.customEvent;
    /** @type {?} */
    DynamicMaterialFormGroupComponent.prototype.focus;
    /** @type {?} */
    DynamicMaterialFormGroupComponent.prototype.layoutService;
    /** @type {?} */
    DynamicMaterialFormGroupComponent.prototype.validationService;
}
